### PR TITLE
Bug 1853138/Bug 1886954 - Switch to Email::Sender and Email::Address::XS

### PR DESCRIPTION
**Pull request by mrenvoize**
This patch updates us from using Email::Address to Email::Address::XS to resolve CVE-2015-7686.

Regex comparisons have been replaced by parse + is_valid calls as per current documentation.

#### Additional info
* [bmo#1853138](https://bugzilla.mozilla.org/show_bug.cgi?id=1853138)

### CODE REVIEW

**Comment by mrenvoize** [comment]
To not require Email::Address::XS independently, I've cherry-picked and rebased the patches migrating from Email::Send to Email::Sender into this branch.

**Comment by mrenvoize** [comment]
I'm going to need a little help getting the tests passing though.. looks like we use a separate docker build for perl dependencies that doesn't respect changes to cpanfile here.. I'll need someone to explain how your workflow works to be able to get those updated dependencies pulled in prior to the tests being able to run as expected.

**Comment by mrenvoize** [comment]
Squashed a couple of minor fixes highlighted by failing tests on the corresponding 5.2 pull request.

**Review by justdave — CHANGES_REQUESTED** [review]
I just ran the tests and they're all failing. Looks like a missing dependency that didn't get caught. (Email::Address::Simple?)

**Comment by mrenvoize** [comment]
> I just ran the tests and they're all failing. Looks like a missing dependency that didn't get caught. (Email::Address::Simple?)

I believe I resolved these as soon as they were reported, hopefully, tests should all be passing now once given the opportunity to run.  I've rebased the branch now the CircleCI updates have landed too, thanks for working on that :)

**Comment by mrenvoize** [comment]
Can we run the tests here please..

File: Makefile.PL [hunk 1]
```diff
@@ -58,7 +58,7 @@ my %requires = (
   'Digest::SHA'                => '5.47',
   'EV'                         => '4.0',
   'Email::MIME'                => '1.904',
-  'Email::Send'                => '1.911',
+  'Email::Sender'              => '2.600',
```
Comment (justdave) [inline Makefile.PL:61]: Given the current test failure, I think you need to mention Email::Sender::Simple here, too.

**Review by justdave — CHANGES_REQUESTED** [review]

File: Makefile.PL (hunk 1 above)
Comment (justdave) [reply Makefile.PL:61]: Actually, you don't.  Email::Sender::Simple can be removed from cpanfile, too.

Comment (mrenvoize) [reply Makefile.PL:61]: I've dropped that commit now :)

**Review by justdave — APPROVED** [review]

**Pull request merged**